### PR TITLE
[fix#2055][config] optimize configuration initialization function  and add endpoint debug log

### PR DESCRIPTION
**Pull request by SnowMoon-Dev**
optimize configuration initialization function  and add endpoint debug log

close [#2055](https://github.com/apache/incubator-devlake/issues/2055)

## Verify this pull request

Manually verified the change by testing locally.

### CODE REVIEW

**Comment by klesh** [comment]
Hi， Can you provide some screen-shot of the logging output?

**Comment by SnowMoon-Dev** [comment]
> Hi， Can you provide some screen-shot of the logging output?

image
image

self-test didn't find any problems

**Comment by matrixji** [comment]
Short question, currently we using **ENV_PATH=/app/config/.env** in helm deployment, does this PR requires we change **ENV_PATH** to **/app/config** ? If so, we could adapt it after this PR merged.

File: config/config_test.go [hunk 1]
```diff
@@ -18,57 +18,42 @@ limitations under the License.
...
+	file, _ := fs.Create(filename)
+	defer file.Close()
+	_ = WriteConfigAs(v, filename)
+	isEmpty, _ := afero.IsEmpty(fs, filename)
+	assert.False(t, isEmpty)
+	err := fs.Remove(filename)
+	assert.Equal(t, err == nil, true)
 }
 
-func TestWriteConfigToEnvPath(t *testing.T) {
```
Comment (matrixji) [inline config/config_test.go:51]: We'd better not remove the testing code, but adapt it to work for the new implementations.

**Comment by SnowMoon-Dev** [comment]
> Short question, currently we using **ENV_PATH=/app/config/.env** in helm deployment, does this PR requires we change **ENV_PATH** to **/app/config** ? If so, we could adapt it after this PR merged.

It seems that there is no need to modify, I deployed with HELM yesterday and successfully started the project

**Comment by matrixji** [comment]
> > Short question, currently we using **ENV_PATH=/app/config/.env** in helm deployment, does this PR requires we change **ENV_PATH** to **/app/config** ? If so, we could adapt it after this PR merged.
> 
> It seems that there is no need to modify, I deployed with HELM yesterday and successfully started the project ...

Those changes LGTM, right now. 

And if possible could you help to confirm with the below steps:
- Change some configurations and save.
- Check where whether the configurations are saved to the .env  with the expected path

**Comment by warren830** [comment]
Thank you @SnowMoon-Dev @matrixji , you are super hero for us!

**Review by warren830 — APPROVED** [review]
LGTM

**Comment by warren830** [comment]
> > > Short question, currently we using **ENV_PATH=/app/config/.env** in helm deployment, does this PR requires we change **ENV_PATH** to **/app/config** ? If so, we could adapt it after this PR merged.
> > 
> > 
> > It seems that there is no need to modify, I deployed with HELM yesterday and successfully started the project ...
> 
> Those changes LGTM, right now.
> 
> And if possible could you help to confirm with the below steps:
> 
> * Change some configurations and save.
> * Check where whether the configurations are saved to the .env  with the expected path

I deployed through helm, and it works very well, thank you Bin @matrixji ! And thank you Shimin @SnowMoon-Dev !

**Pull request merged**